helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl

```diff
@@ -73,13 +73,9 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 Create the name of the service account to use
 */}}
 {{- define "dagsterUserDeployments.serviceAccountName" -}}
-{{- $serviceAccountName := "" }}
-
-{{- if .Values.global }}
```

Comment:
So one interesting note here. Accessing `.Values.global.serviceAccountName` fails if `.Values.global` is not defined. That's presumably why this condition is here.

But, if you pass `.Values.global` through `| default dict`, you can then access `.serviceAccountName` without failure -- it'll just be empty.

So, in order to support `.Values.global.serviceAccountName` with a default to `.Values.serviceAccount.name`, I rewrote this whole stanza.

The priority order now is:
1. `.Values.global.serviceAccountName`
2. `.Values.serviceAccount.name`
3. `printf "%s-user-deployments"`